client/blocks/subscription-length-picker/option.jsx

```diff
@@ -0,0 +1,126 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { uniqueId } from 'lodash';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+import Badge from 'components/badge';
+import { TERM_ANNUALLY, TERM_BIENNIALLY, TERM_MONTHLY } from 'lib/plans/constants';
+
+export class SubscriptionLengthOption extends React.Component {
+	static propTypes = {
+		term: PropTypes.string.isRequired,
+		savePercent: PropTypes.number.isRequired,
+		price: PropTypes.string.isRequired,
+		pricePerMonth: PropTypes.string.isRequired,
+		checked: PropTypes.bool.isRequired,
+		value: PropTypes.any.isRequired,
+		onCheck: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		checked: false,
+		savePercent: 0,
+		onCheck: () => null,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.htmlId = uniqueId( 'subscription-option-' );
+	}
+
+	render() {
+		const { savePercent, price, term, checked } = this.props;
+		const className = classnames( 'subscription-length-picker__option', {
+			'is-active': checked,
+		} );
+		return (
+			<label className={ className } htmlFor={ this.htmlId }>
+				<div className="subscription-length-picker__option-radio-wrapper">
+					<input
+						id={ this.htmlId }
+						type="radio"
+						className="subscription-length-picker__option-radio"
+						checked={ checked }
+						onChange={ this.handleChange }
+					/>
+				</div>
+
+				<div className="subscription-length-picker__option-content">
+					<div className="subscription-length-picker__option-header">
+						<div className="subscription-length-picker__option-term">{ this.getTermText() }</div>
+						<div className="subscription-length-picker__option-discount">
+							{ savePercent ? this.renderSaveBadge() : false }
```

Comment:
are you opposed to unlinking these functions here? I don't think that extracting them into separate functions is helping us due to the additional redirection without saving on logic/complexity/linecount